packages/middleware/src/loggerManager.ts

```diff
@@ -0,0 +1,108 @@
+import type { Response } from "express";
+import { LoggerInterface } from "@vue-storefront/logger";
+import type {
+  AlokaiContainer,
+  IntegrationContext,
+  MiddlewareConfig,
+} from "./types";
+
+type LoggerInstance = LoggerInterface;
+const GLOBAL_MIDDLEWARE_CFG_KEY = Symbol("GLOBAL_MIDDLEWARE_CFG_KEY");
+export class LoggerManager<
+  TLoggerConfig = any,
+  TIntegrationContext extends Record<string, IntegrationContext> = any
+> {
+  private instances: Record<string | symbol, LoggerInstance> = {};
+
+  /** Creating minimal amount of needed Logger instances * */
+  constructor(
+    config: MiddlewareConfig<TIntegrationContext>, // Whole middleware.config.js object
+    buildLogger: (logger: TLoggerConfig) => LoggerInstance
+  ) {
+    // Global Logger
+    this.instances[GLOBAL_MIDDLEWARE_CFG_KEY] = buildLogger(
+      config.logger as TLoggerConfig // logger is on same level as integrations
+    );
+    // Logger for every integration having custom logger config
+    for (const [integrationName, integrationEntry] of Object.entries(
+      config.integrations
+    )) {
+      if (this.hasCustomLoggerConfig(integrationEntry)) {
+        // every integration inside `configuration` object can have own logger too
+        this.instances[integrationName] = buildLogger(
+          integrationEntry.logger as TLoggerConfig
+        );
+      }
+    }
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  private hasCustomLoggerConfig(integrationEntry) {
+    return Boolean(integrationEntry.logger);
+  }
+
+  // getting reference to instance of logger by integration tag name
+  public get(integrationName) {
+    return (
+      this.instances[integrationName] ||
+      this.instances[GLOBAL_MIDDLEWARE_CFG_KEY]
+    );
+  }
+
+  // getting reference to global instance of logger
+  public getGlobal() {
```

Comment:
why do we need this?, just make get(integrationName?: string) and keep the rest, if you will not pass an arg it will get the global, there is no point to have a separate public method.